kms/pkcs11/pkcs11.go

```diff
@@ -157,6 +157,27 @@ func (k *PKCS11) CreateSigner(req *apiv1.CreateSignerRequest) (crypto.Signer, er
 	return signer, nil
 }
 
+// CreateDecrypter creates a decrypter using a key present in the PKCS#11
+// module.
+func (k *PKCS11) CreateDecrypter(req *apiv1.CreateDecrypterRequest) (crypto.Decrypter, error) {
+	if req.DecryptionKey == "" {
+		return nil, errors.New("createDecrypterRequest 'decriptionKey' cannot be empty")
+	}
+
+	signer, err := findSigner(k.p11, req.DecryptionKey)
```

Comment:
It may be good to introduce a `findDecrypter`, which could wrap the type assertions. Not a blocker, IMO.

Reply:
I don't think is worthy because at the end it will duplicate findSigner or call findSigner too.

Reply:
Agree, not required.